File: training-data/elixir/anagram_check.exs
defmodule AnagramCheck do
  def anagram?(a, b) do
    normalize(a) == normalize(b)
  end

  defp normalize(str) do
    str
    |> String.downcase()
    |> String.replace(~r/\s+/, "")
    |> String.graphemes()
    |> Enum.sort()
  end
end

IO.inspect(AnagramCheck.anagram?("listen", "silent"))
IO.inspect(AnagramCheck.anagram?("hello", "world"))

File: training-data/zig/anagram_check.zig
const std = @import("std");
const allocator = std.heap.page_allocator;

fn normalize(s: []const u8) ![]u8 {
    var list = std.ArrayList(u8).init(allocator);
    for (s) |c| {
        if (c != ' ') try list.append(std.ascii.toLower(c));
    }
    std.mem.sort(u8, list.items, {}, std.sort.asc(u8));
    return list.toOwnedSlice();
}

fn isAnagram(a: []const u8, b: []const u8) !bool {
    const na = try normalize(a);
    defer allocator.free(na);
    const nb = try normalize(b);
    defer allocator.free(nb);
    return std.mem.eql(u8, na, nb);
}

pub fn main() !void {
    std.debug.print("{}\n", .{try isAnagram("listen", "silent")});
    std.debug.print("{}\n", .{try isAnagram("hello", "world")});
}
